src/Scaffolding/VS.Web.CG.Mvc/Blazor/BlazorWebCRUDGenerator.cs

```diff
@@ -0,0 +1,352 @@
+            {
+                throw new InvalidOperationException("Unable to get model and/or dbcontext metadata.");
+            }
+
+            if (!string.IsNullOrEmpty(modelTypeAndContextModel.DbContextFullName) && CalledFromCommandline)
+            {
+                EFValidationUtil.ValidateEFDependencies(ProjectContext.PackageDependencies, model.DatabaseProvider);
+            }
+
+            var templateModel = new BlazorModel(modelTypeAndContextModel.ModelType, modelTypeAndContextModel.DbContextFullName)
+            {
+                Namespace = modelTypeAndContextModel.ModelType.Namespace,
+                ModelMetadata = modelTypeAndContextModel.ContextProcessingResult?.ModelMetadata,
+                DatabaseProvider = model.DatabaseProvider,
+                Template = model.TemplateName
+            };
+
+            ExecuteTemplates(templateModel);
+            await ModifyProgramCsAsync();
+        }
+
+        internal IList<string> GetT4Templates(string templateName)
+        {
+            var templates = new List<string>();
+            var crudTemplate = templateName.Equals("crud", StringComparison.OrdinalIgnoreCase);
+            if (crudTemplate)
+            {
+                return CRUDTemplates.Values.ToList();
+            }
+            else if (CRUDTemplates.TryGetValue(templateName, out var t4Template))
+            {
+                templates.Add(t4Template);
+            }
+            else
+            {
+                Logger.LogMessage($"Invalid template for the Blazor CRUD scaffolder '{templateName}' entered!", LogMessageLevel.Error);
+                throw new ArgumentException(templateName);
+            }
+
+            return templates;
+        }
+
+        internal string ValidateAndGetOutputPath(string modelName, string templateName, string relativeFolderPath = null)
+        {
+            string outputFileName = string.IsNullOrEmpty(modelName) ?
+                $"{templateName}{Constants.BlazorExtension}" :
+                Path.Combine($"{modelName}Pages", $"{templateName}{Constants.BlazorExtension}");
+            string outputFolder = string.IsNullOrEmpty(relativeFolderPath) ?
+                Path.Combine(AppInfo.ApplicationBasePath, "Components", "Pages") :
+                Path.Combine(AppInfo.ApplicationBasePath, relativeFolderPath);
+
+            var outputPath = Path.Combine(outputFolder, outputFileName);
+            return outputPath;
+        }
+
+        internal async Task ModifyProgramCsAsync()
+        {
+            var jsonText = GetBlazorCodeModifierConfig();
+            CodeModifierConfig minimalApiChangesConfig = null;
+            try
+            {
+                minimalApiChangesConfig = JsonSerializer.Deserialize<CodeModifierConfig>(jsonText);
+            }
+            catch (JsonException ex)
+            {
+                ConsoleLogger.LogMessage($"Error deserializing blazorWebCrudChanges.json file. {ex.Message}");
```

Comment:
If possible, I'd add the whole file name with the path.

Reply:
sure yeah, sg